Improve write performance

niksajakovljevic opened the pull request:
For normalized table setup we skip using DB triggers (existing triggers in pg_prometheus appear to be slow), but write first into short lived temporary table and then inserting data into labels and values tables from there. Temporary table data is cleaned on each commit. Refactoring db tests to use the actual client creation logic.

niksajakovljevic committed:
Improve write performance

For normalized table setup we skip using DB triggers (existing triggers in pg_prometheus appear to be slow) but write first into short lived temporary table and then inserting data into labels and values tables from there. Temporary table data is cleaned on each commit. Refactoring db tests to use the actual client creation logic.

RobAtticus commented on postgresql/client.go:
```diff
@@ -51,7 +51,7 @@ func ParseFlags(cfg *Config) *Config {
 	flag.StringVar(&cfg.sslMode, "pg.ssl-mode", "disable", "The PostgreSQL connection ssl mode")
 	flag.StringVar(&cfg.table, "pg.table", "metrics", "The PostgreSQL table")
-	flag.StringVar(&cfg.copyTable, "pg.copy-table", "metrics_copy", "The PostgreSQL table")
+	flag.StringVar(&cfg.copyTable, "pg.copy-table", "", "The PostgreSQL table")
```
These flags have identical descriptions so its a bit confusing what they do

RobAtticus commented on postgresql/client.go:
```diff
@@ -68,6 +68,15 @@ type Client struct {
 	cfg *Config
 }
 
+const sqlCreateTmpTable = "CREATE TEMPORARY TABLE IF NOT EXISTS %s_tmp(sample prom_sample) ON COMMIT DELETE ROWS;"
```
change to a const grouping?:
```
const (
  sqlCreateTmpTable = ...
  ...
)
```

RobAtticus commented on postgresql/client.go:
```diff
@@ -101,6 +110,7 @@ func NewClient(cfg *Config) *Client {
 		os.Exit(1)
 	}
 
+	createTmpTableStmt, err = db.Prepare(fmt.Sprintf(sqlCreateTmpTable, cfg.table))
```
`err` not checked or used here?

niksajakovljevic committed:
Minor fixes

Adjust usage messages. Log error and exit on failure to prepare create tmp table statement

davidkohn88 reviewed (approved):
SQL looks fine.
